api/rosetta/data_integration_test.go

```diff
@@ -53,7 +55,11 @@ func setupDB(t *testing.T) *badger.DB {
 	db, err := badger.Open(opts)
 	require.NoError(t, err)
 
-	reader := hex.NewDecoder(strings.NewReader(snapshots.Rosetta))
+	dict, _ := hex.DecodeString(dictionaries.Payload)
+	reader, err := zstd.NewReader(hex.NewDecoder(strings.NewReader(snapshots.Rosetta)),
+		zstd.WithDecoderDicts(dict),
+	)
```

Comment:
I'd suggest either

```suggestion
	reader, err := zstd.NewReader(
		hex.NewDecoder(strings.NewReader(snapshots.Rosetta)),
		zstd.WithDecoderDicts(dict),
	)
```

Or declaring a variable with `hex.NewDecoder(strings.NewReader(snapshots.Rosetta))` as its value and using that.